### PR TITLE
test(tile, switch): update cypress tests with a new URL approach

### PR DESCRIPTION
**Pull request by KatarzynaQA**
### Proposed behaviour

Apply new `url` approach in tests for:
`tile` (timing: was `1:00` -> `0:22` ),
`switch` (timing: `1:02` ).

### Current behaviour

Currently we are passing all parameters via `knobs`.

### Checklist

- [x] Commits follow our style guide
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: cypress/features/regression/experimental/checkbox.feature [hunk 1]
```diff
@@ -32,12 +32,12 @@ Feature: Experimental Checkbox component
...
   Scenario: Enable fieldHelpInline
     When I open default "Experimental Checkbox" component in noIFrame with "checkbox" json from "experimental" using "fieldHelpInlineDisabled" object name
-    Then Checkbox is set to fieldHelpInline and has margin-left set to "24px"
+    Then Checkbox is set to fieldHelpInline and has marginLeft set to "24px"
 
   @positive
   Scenario: Enable and disable fieldHelpInline
     When I open default "Experimental Checkbox" component in noIFrame with "checkbox" json from "experimental" using "fieldHelpInline" object name
-    Then Checkbox is not set to fieldHelpInline and has margin set to "0px"
+    Then Checkbox is not set to fieldHelpInline and has marginTop se to "0px"
```
Comment (DlgSHi) [inline cypress/features/regression/experimental/checkbox.feature:40]:
```suggestion
    Then Checkbox is not set to fieldHelpInline and has marginTop set to "0px"
```

File: cypress/features/regression/experimental/switch.feature [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  @positive
+  Scenario: Enable fieldHelpInline
+    When I open default "Experimental Switch" component in noIFrame with "switch" json from "experimental" using "fieldHelpInline" object name
+    Then Switch is set to fieldHelpInline and has marginLeft set to "0px"
+
+  @positive
+  Scenario: Disable fieldHelpInline
+    When I open default "Experimental Switch" component in noIFrame with "switch" json from "experimental" using "fieldHelpInlineFalse" object name
+    Then Switch is not set to fieldHelpInline and has marginTop se to "8px"
```
Comment (DlgSHi) [inline cypress/features/regression/experimental/switch.feature:23]:
```suggestion
    Then Switch is not set to fieldHelpInline and has marginTop set to "8px"
```

File: cypress/features/regression/tile.feature [hunk 3]
```diff
@@ -1,55 +1,52 @@
...
+      | pixelWidth | nameOfObject   |
+      | 1          | pixelWidth1    |
+      | 100        | pixelWidth100  |
+      | 1999       | pixelWidth1999 |
 
   @positive
   Scenario Outline: Change Tile width to <width>
-    When I set group Default width slider to <width>
+    When I open default "Tile" component in noIFrame with "tile" json from "commonComponents" using "<nameOfObject>" object name
     Then Tile width is set to "<width>"
```
Comment (DlgSHi) [inline cypress/features/regression/tile.feature:47]:
```suggestion
    Then Tile width is set to <width>
```
shouldn't it be changed to `int`?

File: cypress/fixtures/commonComponents/tile.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
+{
+  "asTile": {
+    "as_Default": "tile"
+  },
+  "asTransparent": {
+    "as_Default": "transparent"
+  },
+  "orientationHorizontal": {
+  "orientation_Default": "horizontal"
```
Comment (DlgSHi) [inline cypress/fixtures/commonComponents/tile.json:9]: `json` format

File: cypress/locators/tile/locators.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 // knobs locators
 export const PIXEL_WIDTH_SLIDER = 'input[name="pixelWidth"]';
 export const WIDTH_SLIDER = 'input[name="width"]';
-
-// component preview locators
-export const TILE = 'div[data-component="tile"]';
+export const TILE = '[data-component="tile"]';
+export const TILE_BODY = '[data-element="body"]';
+export const TILE_TITLE = '[data-element="title"]';
```
Comment (DlgSHi) [inline cypress/locators/tile/locators.js:6]: do we need it here ? if we could use `getDataElementByValue('title')`

File: cypress/locators/tile/locators.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 // knobs locators
 export const PIXEL_WIDTH_SLIDER = 'input[name="pixelWidth"]';
```
Comment (DlgSHi) [inline cypress/locators/tile/locators.js:2]: we are not using knobs locators any more - please remove those

File: cypress/locators/tile/locators.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
 export const WIDTH_SLIDER = 'input[name="width"]';
```
Comment (DlgSHi) [inline cypress/locators/tile/locators.js:3]: we are not using knobs locators any more - please remove those

File: cypress/support/step-definitions/common-steps.js [hunk 8]
```diff
@@ -270,6 +270,17 @@ Then('fieldHelp on preview is set to {word} in NoIFrame', (text) => {
   fieldHelpPreviewNoIFrame().should('have.text', text);
 });
 
+Then('{word} is set to fieldHelpInline and has marginLeft set to {string}', (componentName, marginLeft) => {
+  fieldHelpPreviewNoIFrame().should('have.css', 'margin-left', marginLeft)
+    .and('have.css', 'margin-top', '0px')
+    .and('have.css', 'padding-left', '0px');
+});
+
+Then('{word} is not set to fieldHelpInline and has marginTop se to {string}', (componentName, marginTop) => {
```
Comment (DlgSHi) [inline cypress/support/step-definitions/common-steps.js:279]: typo 
```suggestion
Then('{word} is not set to fieldHelpInline and has marginTop set to {string}', (componentName, marginTop) => {
```

File: cypress/support/step-definitions/switch-steps.js [hunk 9]
```diff
@@ -1,51 +1,46 @@
...
+    .and('not.have.attr', 'disabled');
 });
 
-Then('Switch component is not set to fieldHelpInline', () => {
-  switchProperties().should('not.have.class', 'carbon-checkbox__help-text--inline');
+Then('Switch is set to {string}', (size) => {
+  if (size === 'small') {
+    switchInput().should('have.css', 'width', '60px')
+      .and('have.css', 'height', '24px');
+  } else {
```
Comment (DlgSHi) [inline cypress/support/step-definitions/switch-steps.js:24]:
```suggestion
  } else if (size === 'large') {
```

File: cypress/support/step-definitions/tile-steps.js [hunk 10]
```diff
@@ -20,11 +19,3 @@ Then('Tile width is set to {string}', (width) => {
 Then('Tile pixel width is set to {string}', (width) => {
```
Comment (DlgSHi) [inline cypress/support/step-definitions/tile-steps.js:19]:
```suggestion
Then('Tile pixel width is set to {int}', (width) => {
```

File: cypress/locators/tile/locators.js (hunk 5 above)
Comment (KatarzynaQA) [reply cypress/locators/tile/locators.js]: we don't need it. Removed

**Review by DlgSHi — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**